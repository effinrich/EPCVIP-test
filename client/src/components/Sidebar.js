import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import { Box, Button, Layer, ResponsiveContext } from 'grommet'
import { FormClose } from 'grommet-icons'

import MenuButton from './MenuButton'

const Sidebar = ({ items = [], onToggleSidebar, onLogoutClick, ...rest }) => {
  const size = useContext(ResponsiveContext)

  const SidebarContent = () => (
    <Box fill background="#555">
      {size === 'small' && (
        <Box align="end" background="#372F30">
          <Button
            icon={<FormClose size="medium" />}
            onClick={onToggleSidebar}
          />
        </Box>
      )}
      {items.map(({ active, exact, label, path }) => (
        <MenuButton
          active={active}
          exact={exact}
          path={path}
          label={label}
          key={label}
        />
      ))}
      <MenuButton onClick={() => onLogoutClick()} label="Logout" />
    </Box>
  )

  return (
    <div>
      {size === 'small' ? (
        <Layer {...rest}>
          <SidebarContent />
        </Layer>
      ) : (
        <Box
          fill="vertical"
          width="small"
          background="#555"
          elevation="xsmall"
          {...rest}
        >
          <SidebarContent />
        </Box>
      )}
    </div>
  )
}

Sidebar.propTypes = {
  items: PropTypes.array,
  onToggleSidebar: PropTypes.func,
  onLogoutClick: PropTypes.func
}

export default Sidebar
