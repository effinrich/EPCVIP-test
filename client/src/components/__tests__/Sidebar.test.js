import React from 'react'

import { render } from 'utils/testUtils/helperRtl'

import Sidebar from '../Sidebar'

describe('Sidebar Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<Sidebar />)
    expect(container).toMatchSnapshot()
  })

  it('should render child element', () => {
    const { container } = render(<Sidebar />)
    expect(container.firstChild).toBeDefined()
  })
})
