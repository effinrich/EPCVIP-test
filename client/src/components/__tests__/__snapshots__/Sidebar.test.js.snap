// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should render without crashing 1`] = `
<div>
  <div>
    <div
      class="StyledBox-sc-13pk1d4-0 XdbHb"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 lfOCqA"
      >
        <button
          class="StyledButton-sc-323bzc-0 eHYnCT"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kBTLoi"
          >
            <span
              class="StyledText-sc-1sadyjn-0 DkZvS"
            >
              Logout
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
